dotnet/src/Agents/OpenAI/OpenAIAssistantAgent.cs

```diff
@@ -38,57 +40,55 @@ public sealed class OpenAIAssistantAgent : KernelAgent
     public KernelArguments? Arguments { get; init; }
 
     /// <summary>
-    /// A list of previously uploaded file IDs to attach to the assistant.
+    /// The assistant definition.
     /// </summary>
-    public IReadOnlyList<string> FileIds => this._assistant.FileIds;
+    public OpenAIAssistantDefinition Definition { get; private init; }
 
     /// <summary>
-    /// A set of up to 16 key/value pairs that can be attached to an agent, used for
-    /// storing additional information about that object in a structured format.Keys
-    /// may be up to 64 characters in length and values may be up to 512 characters in length.
+    /// Set when the assistant has been deleted via <see cref="DeleteAsync(CancellationToken)"/>.
+    /// An assistant removed by other means will result in an exception when invoked.
     /// </summary>
-    public IReadOnlyDictionary<string, string> Metadata => this._assistant.Metadata;
+    public bool IsDeleted { get; private set; }
 
     /// <summary>
-    /// Expose predefined tools.
+    /// Defines polling behavior for run processing
     /// </summary>
-    internal IReadOnlyList<ToolDefinition> Tools => this._assistant.Tools;
+    public RunPollingOptions PollingOptions { get; } = new();
 
     /// <summary>
-    /// Set when the assistant has been deleted via <see cref="DeleteAsync(CancellationToken)"/>.
-    /// An assistant removed by other means will result in an exception when invoked.
+    /// Expose predefined tools for run-processing.
     /// </summary>
-    public bool IsDeleted { get; private set; }
+    internal IReadOnlyList<ToolDefinition> Tools => this._assistant.Tools;
 
     /// <summary>
     /// Define a new <see cref="OpenAIAssistantAgent"/>.
     /// </summary>
     /// <param name="kernel">The <see cref="Kernel"/> containing services, plugins, and other state for use throughout the operation.</param>
-    /// <param name="config">Configuration for accessing the Assistants API service, such as the api-key.</param>
+    /// <param name="provider">Configuration for accessing the API service.</param>
```

Comment:
Suggesting updating the parameter name to make clear what is this provider about.

```suggestion
    /// <param name="clientProvider">OpenAI client provider for accessing the API service.</param>
```